packages/core/src/shared/errors.ts

```diff
@@ -911,3 +912,30 @@ export function getReasonFromSyntaxError(err: Error): string | undefined {
 
     return undefined
 }
+
+/**
+ * Run a function and swallow any network errors that are thrown inside.
+ * Other types of errors are still thrown.
+ */
+export function runIgnoreNetError<T>(fn: () => T, logMsg?: string): T | undefined
+export function runIgnoreNetError<T>(fn: () => Promise<T>, logMsg?: string): Promise<T> | undefined
+export function runIgnoreNetError<T>(fn: () => T | Promise<T>, logMsg?: string): T | Promise<T | void> | undefined {
```

Comment:
not really sure about this pattern, but to avoid copies of it we could parameterize the "ignore" decision:
```suggestion
export function runIgnoreNetError<T>(fn: () => T | Promise<T>, shouldThrow: (err: Error) => boolean, logMsg?: string): T | Promise<T | void> | undefined {
```